### PR TITLE
JAMES-3061 Cassandra keyspace creation tests

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+
+@ExtendWith(DockerCassandraExtension.class)
+class ResilientClusterProviderTest {
+
+    private final static String TESTING_KEYSPACE = "resilient_cluster_provider_test";
+
+    @Nested
+    class WhenAllowCreatingKeySpace {
```
Comment (Arsnael) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: What about tests when the keyspace creation is disabled?

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                .isTrue();
+        }
+
+        private boolean keyspaceExist(String keyspaceName) {
+            try (Session cassandraSession = cluster.newSession()) {
+                long numberOfKeyspaces = cassandraSession
+                    .execute("SELECT COUNT(*) FROM system_schema.keyspaces where keyspace_name = '" + keyspaceName +"'")
+                    .one()
+                    .getLong("count");
+                if (numberOfKeyspaces > 1 || numberOfKeyspaces < 0) {
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: if (numberOfKeyspaces != 1)

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    throw new IllegalStateException("unexpected keyspace('" + keyspaceName + "') count being " + numberOfKeyspaces);
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: Use string interpolation

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                .doesNotThrowAnyException();
+
+            assertThat(keyspaceExist(TESTING_KEYSPACE))
+                .isTrue();
+        }
+
+        private boolean keyspaceExist(String keyspaceName) {
+            try (Session cassandraSession = cluster.newSession()) {
+                long numberOfKeyspaces = cassandraSession
+                    .execute("SELECT COUNT(*) FROM system_schema.keyspaces where keyspace_name = '" + keyspaceName +"'")
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: Use string interpolation

**Comment by trantienduchn** [comment]
> but I'm curious to know why you decided to write such tests...

I encountered an unexpected cassandra exception when:
 - the keyspace is already created
 - the cassandra connection using dedicated account try to execute the `CREATE KEYSPACE IF EXIST...`
 - I think cassandra ACL prevents the user to execute `CREATE KEYSPACE` regardless the `IF EXIST` afterward...

I want to test case, with a dedicated account, james will check keyspace existent first, then create if necessary instead of using the CREATE KEYSPACE IF EXIST...

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+            assertThatCode(() -> new ResilientClusterProvider(clusterConfiguration))
+                .doesNotThrowAnyException();
+
+            assertThat(keyspaceExist(TESTING_KEYSPACE))
+                .isTrue();
+        }
+
+        private boolean keyspaceExist(String keyspaceName) {
+            try (Session cassandraSession = cluster.newSession()) {
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: why don't you use the query builder?

Comment (trantienduchn) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: I was in hurry

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java (hunk 2 above)
Comment (trantienduchn) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java]: 0 is a valid number (no keyspace had been created)

**Comment by trantienduchn** [comment]
moved to #3131

**Pull request closed without merging**